### DISCUSSION TITLE
Solving an Euler-Lagrange equation to obtain vacancy profile

### DISCUSSION THREAD

**Discussion by rayaprolu143** [idaholab/moose#18286] [Q&A Modules: Phase field] [answered]
Hi,

I am trying to replicate the concentration profile of vacancies for critical nuclei using the expressions in Semenov and Woo paper. I have attached the paper below for reference. By solving Eqn. (9) in 1D and choosing the appropriate dirichlet boundary conditions for either end of the domain as shown in the input file, I was able to obtain vacancy profiles for different supersaturations. However, I could only solve the problem when the bulk free energy expression is in a polynomial form and couldn't do with the free energy expression given in the paper (see Eqn 20) because of the logarithm terms. I did specify in the kernel to use a Taylor expansion form for the logarithms when the value is less than 1e-04  and the source file is attached below. I need help in solving for the vacancy profile when a free energy expression with logarithms is considered.

I did see satisfactory results when I considered polynomial form, as it clearly shows that the critical nuclei decrease with an increase in supersaturation. The image is attached below and the arrow points to an increase in supersaturation of vacancies. Also, it would also be very helpful  if anyone can tell me how to solve the same problem in spherical coordinates.

[Semenov, Woo - 2012 - Phase-field modeling of void formation and growth under irradiation(3).pdf](https://github.com/idaholab/moose/files/6804022/Semenov.Woo.-.2012.-.Phase-field.modeling.of.void.formation.and.growth.under.irradiation.3.pdf)
[ADChemicalPotential.txt](https://github.com/idaholab/moose/files/6804033/ADChemicalPotential.txt)
[Steady-state_nucleation.txt](https://github.com/idaholab/moose/files/6804035/Steady-state_nucleation.txt)
![concentration_profile](https://user-images.githubusercontent.com/14978999/125346978-5b9aca00-e328-11eb-8fdd-72424775baa3.png)

**Comment by GiudGiud** [comment]
Hello

Spherical coordinates should just be setting `coord_type = RSPHERICAL` in [Problem]
As for the logarithmic free energy, what is the issue? Convergence or input?

Guillaume

**Reply by rayaprolu143** [reply]
It is the convergence.

sreekar

**Comment by dschwen** [comment, marked as answer]
Ok, what you did could be more easily done with a `DerivativeParseMaterial` and the `plog` function we provide. The _whole point_ of doing a Taylor expansion for small values is to reduce the stiffness of the equation system by _dropping_ to a lower polynomial order. Your **1000 term Taylor expansion** is basically indistinguishable from the actual logarithm!
